Sourcery/Parsing/Composer.swift

```diff
@@ -112,8 +173,8 @@ struct Composer {
                 tupleElement.type = resolveType(typeName: tupleElement.typeName, containingType: containingType, unique: unique, modules: modules, typealiases: typealiases)
             }
         }
-        return unique[lookupName.unwrappedTypeName]
-            ?? typeFromModule(lookupName.unwrappedTypeName, modules: modules)
+
+        return unique[lookupName.unwrappedTypeName] ?? unique[lookupName.generic?.name ?? ""] ?? typeFromModule(lookupName.unwrappedTypeName, modules: modules)
```

Comment:
That makes me think that `unwrappedTypeName` should consider not just `Optional` type as it does now, but also drop type parameters of any generic types. Then your logic will work here.